src/main/scala/edu/ie3/simona/agent/grid/GridResultsSupport.scala

```diff
@@ -542,51 +541,52 @@ private[grid] trait GridResultsSupport {
   ): (ComparableQuantity[ElectricCurrent], ComparableQuantity[Angle]) =
     (
       iNominal.multiply(iPu.abs).asComparable,
-      angle(iPu)
+      complexToAngle(iPu)
     )
 
-  /** Correct the offset of an angle dependent on the direction. If the
-    * direction is negative, 180° are added
-    */
-  private val offSetCorrection
-      : (ComparableQuantity[Angle], Double) => ComparableQuantity[Angle] =
-    (angle: ComparableQuantity[Angle], dir: Double) =>
-      if (dir < 0) {
-        angle.add(Quantities.getQuantity(180d, PowerSystemUnits.DEGREE_GEOM))
-      } else {
-        angle
-      }
-
-  /** Calculate the angle of the complex value given. The angle has the full
+  /** Calculate the angle of the complex value given. The angle has the proper
     * orientation on the complex plane.
     *
     * @param cplx
     *   The complex value
     * @return
     *   The angle of the complex value
     */
-  private def angle(cplx: Complex): ComparableQuantity[Angle] = cplx match {
-    case _ if cplx.abs == 0 =>
-      /* The complex value has no magnitude, therefore define the angle to zero */
-      Quantities.getQuantity(0d, PowerSystemUnits.DEGREE_GEOM)
-    case Complex(real, imag) =>
-      /* Calculate the angle between real and imaginary part */
-      val baseAngle = atan(imag / real).toDegrees
-
-      if (baseAngle.isNaN) {
-        /* There is only an imaginary part */
-        offSetCorrection(
+  private def complexToAngle(cplx: Complex): ComparableQuantity[Angle] =
+    cplx match {
+      case Complex(0d, 0d) =>
+        /* The complex value has no magnitude, therefore define the angle to zero */
+        Quantities.getQuantity(0d, PowerSystemUnits.DEGREE_GEOM)
+      case Complex(0d, imag) =>
+        /* There is only an imaginary part:
+        Angle can be 90 or 270 degrees, depending on sign of the imaginary part */
+        angleOffsetCorrection(
           Quantities.getQuantity(90d, PowerSystemUnits.DEGREE_GEOM),
           imag
         )
-      } else {
-        /* Correct the angle for full orientation in the complex plane */
-        offSetCorrection(
+      case Complex(real, imag) =>
+        /* Both real and imaginary part are != 0. This means atan can be used to
+        compute an angle between -90 and 90 degrees. To compute the actual angle
+        (between -180 and 180 degrees), 180 degrees have to be added in case that
+        the real part is negative. */
```

Comment:
I think this doesn't quite nail it but feel free to correct me if im wrong.

What we seem to want is the absolute angle of the imaginary part calculated with respect to the positive real axis which relates to a positive real part.
So atan actually computes the direct angle between the imaginary part (Gegenkathete) and the real part (Ankathete). In case the imaginary part is negative we calculate the relative angle with respect to the negative real axis which would mean we have to "turn" the angle by 180°

Reply:
Yes, that's what's happening...
(besides the point that you were probably trying to say "In case the **real** part is negative [...]" in your last sentence)

How would you enhance the commentary here?

Reply:
Maybe like this?
```suggestion
        /* Both real and imaginary part are != 0. This means atan can be used to
        compute an angle between -90 and 90 degrees. To compute the actual angle
        (between -180 and 180 degrees), 180 degrees have to be added in case that
        the real part is negative. */
```

Reply:
I like the proposal

Reply:
Yes, thanks @danielfeismann!